### ISSUE TITLE
[BUG] +javascript-npm-mode not initializing anymore

### ISSUE THREAD

**Issue by sooqua** [hlissner/doom-emacs#1712]
**Describe the issue**
I just reinstalled Arch Linux, and I found that `<SPC> m n` is not working in the same project that it used to. I removed `./node_modules` and reinstalled all the modules, but to no avail. As usual, I tried default configuration before creating this issue.

**Steps to reproduce**
1. Open a javascript nodejs project
2. Press `<SPC> m n`
3. Nothing?
4. Eval `(+javascript-npm-dep-p "react")`
5. Still nothing, even though `package.json` is in the project's root and "react" is in listed in dependencies

**System information**
<details>

```
emacs   version    26.2
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2
        build      Apr 12, 2019
        buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
doom    version    2.0.9
        build      HEAD -> straight, origin/straight ea34b111 2019-08-23 18:57:57 -0400
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /usr/bin/zsh
        uname      Linux 5.2.9-arch1-1-ARCH #1 SMP PREEMPT Fri Aug 16 11:29:43 UTC 2019 x86_64
        path       (/usr/local/bin /usr/bin /bin /usr/local/sbin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/26.2/x86_64-pc-linux-gnu/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion company ivy :ui doom hl-todo modeline nav-flash ophints (popup +all +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces :editor evil file-templates fold multiple-cursors rotate-text snippets :emacs dired electric vc :term eshell :tools eval flycheck (lookup +docsets) magit :lang cc data emacs-lisp javascript lua markdown (org +dragndrop +ipython +pandoc +present) sh web :config default)
        packages   (pkgbuild-mode livid-mode (rainbow-mode :recipe (:host github :repo emacsmirror/rainbow-mode)) js-doc package-lint)
```
</details>

**Comment by sooqua** [comment]
I've bisected the straight branch and [this commit](https://github.com/hlissner/doom-emacs/commit/ee81d0533afb51418f7fced75318e4db74036182) is closest I could get to "the first bad commit", because I'm getting errors on startup.

**Comment by sooqua** [comment]
[This is the last good commit](https://github.com/hlissner/doom-emacs/commit/eb567f51c89dcdfb60498f4e64cc820f66e1669d) that doesn't throw errors on startup and `<SPC> m n` and ESLint is working.

**Comment by robbert-vdh** [comment]
I stumbled upon this a few days ago when the Django template syntax for web-mode was no longer triggering in my Django projects. There's a small typo in the new [`def-project-mode!`](https://github.com/hlissner/doom-emacs/blob/e498a7e2cb66ef1d6305a206a3f3924e4c4501ac/core/core-projects.el#L245) implementation where the hook function name's format string is missing a placeholder. `(format "-in-" mode)` should have been `(format "-in-%s-" mode)`. I then also noticed that the `,(when match ...)` part of the generated function would cause a add `nil` to be added the and form when the `:match` option is not being used. That's what's causing the project minor-modes to not work anymore.

**Comment by sooqua** [comment]
For anyone wondering where are the `<SPC> m n` npm commands, you can now access them with `<C-c> n`

**Comment by hlissner** [comment]
They should still be on `SPC m n`. See https://github.com/hlissner/doom-emacs/blob/develop/modules/lang/javascript/config.el#L244-L246.

**Comment by hlissner** [comment]
Nevermind. `npm-mode` doesn't use the conventional `npm-mode-map` keymap, it uses `npm-mode-keymap`. It's been corrected in 21883d9.
